src/Domain/LeanCode.DomainModels.Generators/IdSource.cs

```diff
@@ -0,0 +1,243 @@
+namespace LeanCode.DomainModels.Generators;
+
+internal static class IdSource
+{
+    public static string Build(TypedIdData data)
+    {
+        switch (data.Format)
+        {
+            case TypedIdFormat.RawInt:
+                return BuildRaw(data, "int", "Int", null, "0", "CultureInfo.InvariantCulture");
+
+            case TypedIdFormat.RawLong:
+                return BuildRaw(data, "long", "Long", null, "0", "CultureInfo.InvariantCulture");
+
+            case TypedIdFormat.RawGuid:
+                return BuildRaw(data, "Guid", "Guid", "Guid.NewGuid()", "Guid.Empty", "");
+
+            case TypedIdFormat.PrefixedGuid:
+                return BuildPrefixed(data, "Guid.NewGuid()", 32, "Guid.Empty", true, "Guid.TryParseExact", "N");
+
+            default:
+                throw new ArgumentException("Unsupported ID format.");
+        }
+    }
+
+    private static string BuildPrefixed(
+        TypedIdData data,
+        string randomValueGenerator,
+        int valueLength,
+        string defaultValue,
+        bool generateFromCtor,
+        string tryParseExact,
+        string rawFormat
+    )
+    {
+        var prefix = data.CustomPrefix?.ToLowerInvariant() ?? GetDefaultPrefix(data.TypeName);
+        var fromCtor = generateFromCtor
+            ? $@"public {data.TypeName}(Guid v) => value = string.Create(null, stackalloc char[RawLength], $""{{TypePrefix}}{{Separator}}{{v:{rawFormat}}}"");"
+            : "";
+        var randomFactory = !data.SkipRandomGenerator
+            ? $@"public static {data.TypeName} New() => new({randomValueGenerator});"
+            : "";
+        return $@"// <auto-generated />
```

Comment:
Wouldn't all this be more readable using `$$"""` multiline strings?

Reply:
I don't think it would change that much, but will try